ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -207,6 +222,38 @@
     return metadataList;
   }
 
+  private Set<String> findSubtopologySourceTopics() {
+    for (final Subtopology subtopology : topology.describe().subtopologies()) {
+      boolean containsStateStore = false;
+      for (final TopologyDescription.Node node : subtopology.nodes()) {
+        if (node instanceof Processor) {
+          final Processor processor = (Processor) node;
+          if (processor.stores().contains(stateStoreName)) {
+            containsStateStore = true;
+          }
+        }
+      }
+
+      if (!containsStateStore) {
+        continue;
+      }
+
+      for (final TopologyDescription.Node node : subtopology.nodes()) {
+        if (node instanceof Source) {
+          final Source source = (Source) node;
+          if (source.topicSet() != null) {
+            return source.topicSet()
+                .stream()
+                .map(name -> applicationId + "-" + name)
```

Comment:
+1. Good catch.

Reply:
If the user chose a name that collided with an internal name, couldn't this incorrectly choose the wrong source topic?  I know it's a bit of a contrived case, but it seems possible.

Reply:
Today we do not forbid that programmatically and if that happens, it would be of a bigger issue: during the first rebalance when we would like to create the intermediate topic, if collides with the existing source topics, then it would throw if the number of partitions do not match expected value, or would assume the topic is already created if they match. And we could potentially have a loop where a task read as source and write as sink the same topic.